src/Tornado.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+
+import {
+  EGroupedBarLayout,
+  IAxes,
+  IChartAdaptor,
+  IChartState,
+  IDomain,
+  IGrid,
+  IHistogramBar,
+  IMargin,
+  IStroke,
+  TipContentFn,
+} from './Histogram';
+import { IGroupDataItem } from './HistogramD3';
+import { tornadoD3 } from './TornadoD3';
+import { DeepPartial } from './utils/types';
+
+export interface ITornadoDataSet {
+  borderColors?: string[];
+  colors?: string[];
+  label: string;
+  data: [number[], number[]];
+}
+
+export interface ITornadoData {
+  bins: string[];
+  counts: ITornadoDataSet[];
+  colorScheme?: string[];
+  title?: string;
+}
+
+export interface ITornadoProps {
+  axis: IAxes;
+  bar: IHistogramBar;
+  center: boolean;
+  className: string;
+  data: ITornadoData;
+  delay: number;
+  duration: number;
+  colorScheme: string[];
+  domain: IDomain;
+  grid: IGrid;
+  height: number;
+  margin: IMargin;
+  groupLayout: EGroupedBarLayout;
+  onClick?: (bar: IGroupDataItem) => void;
+  splitBins: [string, string];
+  stroke: IStroke;
+  tip: any;
+  tipContainer: string;
+  tipContentFn: TipContentFn<string>;
+  visible: { [key: string]: boolean };
+  width: number | string;
+}
+
+/**
+ * Tornado component
+ */
+class Tornado extends Component<DeepPartial<ITornadoProps>, IChartState> {
```

Comment:
Have you looked into making this using hooks? With `useEffect` it could be simplified a fair bit

Reply:
would require rewriting all of the other charts as well - whilst it would be nice its prob not a priority atm